lib/tests/test.js

```diff
@@ -349,48 +346,20 @@ class Test {
     });
   }
 
-  require(module) {
-    if (module.startsWith('Embark/contracts/')) {
-      const contractName = module.substr(17);
-      if (this.contracts[contractName]) {
-        return this.contracts[contractName];
-      }
-      let contract = this.engine.contractsManager.contracts[contractName];
-      if (!contract) {
-        const contractNames = Object.keys(this.engine.contractsManager.contracts);
-        // It is probably an instanceof
-        contractNames.find(contrName => {
-          // Find a contract with a similar name
-          if (contractName.indexOf(contrName) > -1) {
-            contract = this.engine.contractsManager.contracts[contrName];
-            return true;
-          }
-          return false;
-        });
-        // If still nothing, assign bogus one, we will redefine it anyway on deploy
-        if (!contract) {
-          console.warn(__('Could not recognize the contract name "%s"', contractName));
-          console.warn(__('If it is an instance of another contract, it will be reassigned on deploy'));
-          console.warn(__('Otherwise, you can rename the contract to contain the parent contract in the name eg: Token2 for Token'));
-          contract = this.engine.contractsManager.contracts[contractNames[0]];
-        }
-      }
-      this.contracts[contractName] = new EmbarkJS.Contract({
-        abi: contract.abiDefinition,
-        address: contract.address,
-        from: this.web3.eth.defaultAccount,
-        gas: 6000000,
-        web3: this.web3
-      });
-      this.contracts[contractName].address = contract.address;
-      this.contracts[contractName].options.data = contract.code;
-      this.contracts[contractName].options.gas = 6000000;
-      this.web3.eth.getAccounts().then((accounts) => {
-        this.contracts[contractName].options.from = contract.from || accounts[0];
-      });
-      return this.contracts[contractName];
+  require(path) {
+    const prefix = 'Embark/contracts/';
+    if (!path.startsWith(prefix)) {
+      throw new Error(__('Unknown module %s', path));
     }
-    throw new Error(__('Unknown module %s', module));
+    let contractName = path.replace(prefix, "");
+    let contract = this.contracts[contractName];
+    if (contract) {
+      return contract;
+    }
+
+    let newContract = {};
```

Comment:
I'm scarred that if the user enters a bad contract name, he'll get a pretty ugly error. I'll test it myself.